django_cloud_deploy/tests/cleanup/resource_cleanup.py

```diff
@@ -0,0 +1,126 @@
+        create_time = utils.parse_creation_time(resource_name)
+        diff = now - create_time
+        return diff > self.MAX_DIFF
+
+    def delete_expired_clusters(self):
+        container_service = discovery.build(
+            'container',
+            'v1',
+            credentials=self.credentials,
+            cache_discovery=False)
+        for cluster_name in self.list_clusters(container_service):
+            if self._should_delete(cluster_name):
+                self._delete_cluster(cluster_name, container_service)
+
+    def delete_expired_buckets(self):
+        storage_service = discovery.build(
+            'storage',
+            'v1',
+            credentials=self.credentials,
+            cache_discovery=False)
+        for bucket_name in self.list_buckets(storage_service):
+            if self._should_delete(bucket_name):
+                self._delete_bucket(bucket_name, storage_service)
+
+    def delete_expired_sql_instances(self):
+        sqladmin_service = discovery.build(
+            'sqladmin',
+            'v1beta4',
+            credentials=self.credentials,
+            cache_discovery=False)
+        for instance_name in self.list_instances(sqladmin_service):
+            if self._should_delete(instance_name):
+                self._clean_up_sql_instance(instance_name, sqladmin_service)
+
+    def delete_expired_service_accounts(self):
+        iam_service = discovery.build(
+            'iam', 'v1', credentials=self.credentials, cache_discovery=False)
+        for account_email in self.list_service_accounts(iam_service):
+            account_name = account_email.split('@')[0]
+            if self._should_delete(account_name):
+                self._delete_service_account(account_email, iam_service)
+
+    def reset_expired_iam_policy(self):
+        cloudresourcemanager_service = discovery.build(
+            'cloudresourcemanager',
+            'v1',
+            credentials=self.credentials,
+            cache_discovery=False)
+        request = cloudresourcemanager_service.projects().getIamPolicy(
+            resource=self.project_id)
+        policy = request.execute()
+        for binding in policy['bindings']:
+            for member in binding['members']:
+                # Member should look like the follows:
+                # serviceAccount:sa-20190301000243-bu47@aaa.com
+                service_account_name = member.split('@')[0].split(':')[-1]
+                if self._should_delete(service_account_name):
+                    binding['members'].remove(member)
+
+        policy['bindings'] = [b for b in policy['bindings'] if b['members']]
+        body = {'policy': policy}
+        request = cloudresourcemanager_service.projects().setIamPolicy(
+            resource=self.project_id, body=body)
+        request.execute()
+
+    def test_cleanup_resources(self):
+        """This is not a test, but cleans up test related resources."""
+        self.delete_expired_clusters()
```

Comment:
If anyone of these throw an error wouldn all subsequent ones not happen? Even though this is not a test, is it either A) worth breaking up into individual "tests", or try catching each of these?

Reply:
Nice catch! Actually this is similar with what you just told me. The resource cleaning in e2e tests should not raise errors. I will implement this.